### PR TITLE
fix some typos and bugs

### PR DESCRIPTION
**Pull request by chaosma**

### CODE REVIEW

File: circuits/circom/processMessages.circom [hunk 1]
```diff
@@ -295,8 +295,7 @@ template ProcessMessages(
             }
         }
 
-        stateRootQip.leaf <== stateRootHasher.hash;
-        stateLeafQip.root === stateRoots[i + 1];
```
Comment (corydickson) [inline circuits/circom/processMessages.circom:299]: I believe we want this check to make sure the state root proofs include the next state root to be processed

**Comment by corydickson** [comment]
@chaosma Good catch overall, if you can just rebase so the package-lock files don't get changed.

File: circuits/circom/processMessages.circom (hunk 1 above)
Comment (chaosma) [reply circuits/circom/processMessages.circom:299]: the stateLeafQip is not defined and only used here, so I removed it. Is it needed here? stateRootQip.leaf changed to stateRootQip[i].leaf.

Comment (corydickson) [reply circuits/circom/processMessages.circom:299]: Yes it should be set in the top level of the loop

**Pull request merged**